### DISCUSSION TITLE
I can not build main branch

### DISCUSSION THREAD

**Discussion by hendrikebbers** [langchain4j/langchain4j#1503] [Q&A] [answered]
Hi,
Sadly I can not run `mvn install` on my machine. The build always ends with the following errors:

```
[INFO] --- compiler:3.12.1:testCompile (default-testCompile) @ langchain4j ---
[INFO] Recompiling the module because of changed dependency.
[INFO] Compiling 37 source files with javac [debug target 8] to target/test-classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /Users/hendrikebbers/git/langchain4j/langchain4j/src/test/java/dev/langchain4j/service/AiServicesWithRagIT.java:[12,39] cannot find symbol
  symbol:   class AllMiniLmL6V2QuantizedEmbeddingModel
  location: package dev.langchain4j.model.embedding
[ERROR] /Users/hendrikebbers/git/langchain4j/langchain4j/src/test/java/dev/langchain4j/store/embedding/inmemory/InMemoryEmbeddingStoreSerializedTest.java:[4,39] cannot find symbol
  symbol:   class AllMiniLmL6V2QuantizedEmbeddingModel
  location: package dev.langchain4j.model.embedding
[ERROR] /Users/hendrikebbers/git/langchain4j/langchain4j/src/test/java/dev/langchain4j/classification/EmbeddingModelTextClassifierTest.java:[3,39] cannot find symbol
  symbol:   class AllMiniLmL6V2QuantizedEmbeddingModel
  location: package dev.langchain4j.model.embedding
[ERROR] /Users/hendrikebbers/git/langchain4j/langchain4j/src/test/java/dev/langchain4j/store/embedding/inmemory/InMemoryEmbeddingStoreRemovalTest.java:[4,39] cannot find symbol
  symbol:   class AllMiniLmL6V2QuantizedEmbeddingModel
  location: package dev.langchain4j.model.embedding
[ERROR] /Users/hendrikebbers/git/langchain4j/langchain4j/src/test/java/dev/langchain4j/store/embedding/inmemory/InMemoryEmbeddingStoreTest.java:[6,39] cannot find symbol
  symbol:   class AllMiniLmL6V2QuantizedEmbeddingModel
  location: package dev.langchain4j.model.embedding
[INFO] 5 errors 
```

If I ignore test code and do a `mvn compile` I can go some steps further but end with the following errors:

```
[INFO] --- compiler:3.12.1:compile (default-compile) @ langchain4j-easy-rag ---
[INFO] Recompiling the module because of changed dependency.
[INFO] Compiling 1 source file with javac [debug target 8] to target/classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /Users/hendrikebbers/git/langchain4j/langchain4j-easy-rag/src/main/java/dev/langchain4j/data/document/splitter/recursive/RecursiveDocumentSplitterFactory.java:[5,39] cannot find symbol
  symbol:   class HuggingFaceTokenizer
  location: package dev.langchain4j.model.embedding
[ERROR] /Users/hendrikebbers/git/langchain4j/langchain4j-easy-rag/src/main/java/dev/langchain4j/data/document/splitter/recursive/RecursiveDocumentSplitterFactory.java:[12,57] cannot find symbol
  symbol:   class HuggingFaceTokenizer
  location: class dev.langchain4j.data.document.splitter.recursive.RecursiveDocumentSplitterFactory
[INFO] 2 errors 
```

**Comment by mr-cloud** [comment]
`./mvnw clean package -DskipTests` will fail with:
[INFO] BUILD FAILURE                                                                                                                 
[INFO] ------------------------------------------------------------------------                        
[INFO] Total time:  01:11 min                                                                                                        
[INFO] Finished at: 2025-12-04T17:27:12+08:00                                                                                        
[INFO] ------------------------------------------------------------------------                                
[ERROR] Failed to execute goal on project langchain4j: Could not resolve dependencies for project dev.langchain4j:langchain4j:jar:1.1
0.0-SNAPSHOT                                                                                                                         
[ERROR] dependency: dev.langchain4j:langchain4j-embeddings-all-minilm-l6-v2-q:jar:1.10.0-beta17-SNAPSHOT (test)    
[ERROR]         Could not find artifact dev.langchain4j:langchain4j-embeddings-all-minilm-l6-v2-q:jar:1.10.0-beta17-SNAPSHOT                                                                      
[ERROR] dependency: dev.langchain4j:langchain4j-classpath-test-lib:jar:999-SNAPSHOT (test)
[ERROR]         Could not find artifact dev.langchain4j:langchain4j-classpath-test-lib:jar:999-SNAPSHOT

**Reply by dliubarskyi** [reply]
I am currently migrating embeddings into main repo, it should be fixed today or tomorrow. In the meantime, you can checkout https://github.com/langchain4j/langchain4j-embeddings and do `mvn install` locally

**Reply by dliubarskyi** [reply]
Just merged https://github.com/langchain4j/langchain4j/pull/4179, it should be fixed now

**Comment by dliubarskyi** [comment, marked as answer]
Fixed by https://github.com/langchain4j/langchain4j/pull/4179

Feel free to reopen if you still face any issues

**Reply by mr-cloud** [reply]
[ERROR] Failed to execute goal com.googlecode.maven-download-plugin:download-maven-plugin:1.13.0:wget (download-model-file) on project langchain4j-embeddings-bge-small-en: Read timed out -> [Help 1]

can you speed this up?

build from newest commit 778be1b36

**Reply by dliubarskyi** [reply]
> can you speed this up?

What do you mean?

**Reply by dliubarskyi** [reply]
Just commited https://github.com/langchain4j/langchain4j/commit/a06ec71840a7f28dec2882369599115a832f7817, hopefully it will help
